src/mon/OSDMonitor.cc

```diff
@@ -3345,6 +3354,35 @@ void OSDMonitor::take_all_failures(list<MonOpRequestRef>& ls)
   failure_info.clear();
 }
 
+int OSDMonitor::get_grace_interval_threshold()
+{
+  int halflife = g_conf()->mon_osd_laggy_halflife;
+  // Scale the halflife period (default: 1_hr) by
+  // a factor (48 hrs) to calculate the threshold.
+  int grace_threshold_hrs = 48;
+  return halflife * grace_threshold_hrs;
+}
+
+bool OSDMonitor::grace_interval_threshold_exceeded(int last_failed_interval)
+{
+  int grace_interval_threshold_secs = get_grace_interval_threshold();
+  dout(10) << __func__ << " last_failed_interval = " << last_failed_interval
+           << " grace_interval_threshold_secs = " << grace_interval_threshold_secs << dendl;
+  return (last_failed_interval > grace_interval_threshold_secs) ? true : false;
+}
+
+void OSDMonitor::set_default_laggy_params(int target_osd)
+{
+  if (pending_inc.new_xinfo.count(target_osd) == 0) {
+    pending_inc.new_xinfo[target_osd] = osdmap.osd_xinfo[target_osd];
+  }
+  osd_xinfo_t& xi = pending_inc.new_xinfo[target_osd];
+  xi.down_stamp = pending_inc.modified;
+  xi.laggy_probability = 0.0;
+  xi.laggy_interval = 0;
+  dout(20) << " reset laggy, now xi " << xi << dendl;
```

Comment:
nit: might be useful to have `__func__` here too

Reply:
Sure, I omitted it to be consistent with other similar messages (see prepare_boot()).